File: FootballExplorer/app/src/main/java/fr/a42consulting/footballexplorer/Model/PlayersListModel.java
package fr.a42consulting.footballexplorer.Model;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class PlayersListModel {
    public static PlayersListModel fromJson(JSONObject jsonObject) {
        PlayersListModel team = new PlayersListModel();
        try {
            JSONArray jsonArray = jsonObject.getJSONArray("player");

            JSONObject playerJson;
            ArrayList<PlayersListPlayer> players = new ArrayList<>(jsonArray.length());
            // Process each result in json array, decode and convert to player object
            for (int i=0; i < jsonArray.length(); i++) {
                try {
                    playerJson = jsonArray.getJSONObject(i);
                } catch (Exception e) {
                    e.printStackTrace();
                    continue;
                }

                PlayersListPlayer player = PlayersListPlayer.fromJson(playerJson);
                if (player != null) {
                    players.add(player);
                }
            }

            team.players = players;

        } catch (JSONException e) {
            e.printStackTrace();
        }

        return team;
    }
    private ArrayList<PlayersListPlayer> players = new ArrayList<>();

    public ArrayList<PlayersListPlayer> getPlayers() {
        return players;
    }
}

File: FootballExplorer_kotlin/app/src/main/java/fr/a42consulting/footballexplorer/model/PlayersListModel.kt
package fr.a42consulting.footballexplorer.model

import org.json.JSONException
import org.json.JSONObject

import java.util.ArrayList

class PlayersListModel {
    var players = ArrayList<PlayersListPlayer>()
        private set

    companion object {
        fun fromJson(jsonObject: JSONObject): PlayersListModel {
            val team = PlayersListModel()
            try {
                val jsonArray = jsonObject.getJSONArray("player")

                var playerJson: JSONObject
                val players = ArrayList<PlayersListPlayer>(jsonArray.length())
                // Process each result in json array, decode and convert to player object
                for (i in 0 until jsonArray.length()) {
                    try {
                        playerJson = jsonArray.getJSONObject(i)
                    } catch (e: Exception) {
                        e.printStackTrace()
                        continue
                    }

                    val player = PlayersListPlayer.fromJson(playerJson)
                    if (player != null) {
                        players.add(player)
                    }
                }

                team.players = players

            } catch (e: JSONException) {
                e.printStackTrace()
            }

            return team
        }
    }
}
